server.go

```diff
@@ -331,5 +344,5 @@ func (s *Server) ListenAndServe() error {
 
 	s.Logger.Printf("Listening on %v", s.Addr)
```

Comment:
Could this output the wrong address if the `listener` passed into `Serve` is configured to serve on a different address?

If so, what are your thoughts on moving the logger call into `Serve` and `ListenAndServe` to handle each scenario?

Reply:
Great call, will update